Flash: Streamline Point/PublicKey usage & co

omerfirmak opened the pull request:
`FlashAPI` already uses `PublicKey`, prefer `PublicKey` over `Point`

More will probably follow.

omerfirmak committed:
Flash: Streamline Point/PublicKey usage

Prefer PublicKey over Point

omerfirmak committed:
Remove temp initializers

omerfirmak committed:
Flash: Range-ify getManagedChannels

omerfirmak committed:
Flash: simplify handleGossip

omerfirmak committed:
Flash: Avoid allocating while gossiping

Geod24 commented on source/agora/flash/Channel.d:
```diff
@@ -110,7 +110,7 @@ public class Channel
 
     /// @WORKAROUND@: LocalRest issue with timings and sleep not working until
     /// node ctor is done (see Flash restart tests)
```
This has been fixed right ?

omerfirmak replied:
Yep, we fixed that.
But while this works around that, I think it also fixes the race between channel ctor and peer registering itself to the registry. So we can just remove the workaround comments.

Geod24 commented on source/agora/test/Flash.d:
```diff
@@ -436,7 +436,7 @@ public class FlashNodeFactory (FlashListenerType = FlashListener)
     {
         foreach (address; this.addresses)
-            enforce(this.flash_registry.unregister(address.to!string));
+            enforce(this.flash_registry.unregister(PublicKey(address).to!string));
```
Isn't it already a `PublicKey` now ?

omerfirmak replied:
Oops, yes

Geod24 commented on source/agora/test/Flash.d:
```diff
@@ -374,9 +374,9 @@ public class FlashNodeFactory (FlashListenerType = FlashListener)
+        this.addresses ~= PublicKey(pair.V);
         this.nodes ~= api;
-        this.flash_registry.register(pair.V.to!string, api.listener());
+        this.flash_registry.register(PublicKey(pair.V).to!string, api.listener());
```
`this.addresses[$-1].to!string`?

Geod24 commented on source/agora/flash/Node.d:
```diff
@@ -241,10 +241,8 @@ public abstract class FlashNode : FlashControlAPI
-        this.channels[address] = def;
-        DList!ChannelConfig def2;
-        this.pending_channels[address] = def2;
+        this.channels[address] = typeof(this.channels[address]).init;
+        this.pending_channels[address] = typeof(this.pending_channels[address]).init;
```
Why is it here in the first place ? Does it override something ? Everything is default-initialized in D.

omerfirmak replied:
Those are complex types, like AA and DList.
These are there so that when you do `this.channels[address].foo()`, you wont throw.

omerfirmak committed:
Flash: Remove LocalRest workaround related comments

Retry in event loop fixes the race between Channel
contrustion and peer registering itself to the Flash
registry.

Geod24 reviewed (approved):
Me likey
